File: interview-preparation-kit/arrays/new-year-chaos/newYearChaos.cs
static void minimumBribes(int[] q)
        {
            int bribesCount = 0;

            if (isTooChaotic(q))
            {
                Console.WriteLine("Too chaotic");
                return;
            }
            else
            {
                while (findIndexToBribe(q) != -1)
                {
                    q = bribeElements(q, findIndexToBribe(q));
                    bribesCount++;
                }
            }

            Console.WriteLine(bribesCount.ToString());
        }

        static bool isTooChaotic(int[] line)
        {

            for (int i = 0; i < line.Length; i++)
            {

                int indexInLine = Array.IndexOf(line, (i + 1));

                if (isTooFarToReturn(indexInLine, i))
                {
                    return true;
                }
            }

            return false;
        }

        static bool isTooFarToReturn(int indexInLine, int targetIndex)
        {
            if ((targetIndex > indexInLine) && (targetIndex - indexInLine > 2))
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        static int[] bribeElements(int[] line, int index)
        {

            int indexElement = line[index];

            line[index] = line[index + 1];
            line[index + 1] = indexElement;

            return line;
        }

        static int findIndexToBribe(int[] line)
        {
            for (int i = (line.Length - 1); i >= 0; i--)
            {
                if (line[i] != (i + 1))
                {
                    int indexToBribe = Array.IndexOf(line, (i + 1));
                    return indexToBribe;
                }
            }

            return -1;
        }

File: interview-preparation-kit/arrays/new-year-chaos/newYearChaos.js
minimumBribes([2,1,5,3,4]); // 3
minimumBribes([2,5,1,3,4]); // Too chaotic
minimumBribes([5,1,2,3,7,8,6,4]); // Too chaotic
minimumBribes([1,2,5,3,7,8,6,4]); // 7
minimumBribes([1,2,5,3,4,7,8,6]); // 4

function minimumBribes(line) {   
    let bribes = 0;

    line.forEach(function(item, index){
        line[index] = line[index] - 1;
    });

    for(let i = 0; i < line.length; i++){
        if(line[i] - i > 2){
            console.log("Too chaotic");
            return;
        }         
    }

    console.log("Talvez");
}